test/mpi/init/session.c

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
+#include "mpi.h"
+#include <stdio.h>
+#include "mpitest.h"
+
+/* This is example 10.8 from MPI Standard 4.0 */
+
+int errs = 0;
+
+int library_foo_test(void);
+void library_foo_init(void);
+void library_foo_finalize(void);
+
+int main(int argc, char *argv[])
+{
+#ifdef MULT_INIT
+    int rank, size, provided;
+
+    MPI_Init_thread(&argc, &argv, MPI_THREAD_MULTIPLE, &provided);
+    MPI_Comm_size(MPI_COMM_WORLD, &size);
+    MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+    library_foo_test();
+    MPI_Finalize();
+#else
```

Comment:
For `MULT_INIT` test, can we also test calling two `MPI_Session_init`, and maybe need a multithreaded test too?

Reply:
Yeah, probably a good idea

Reply:
mult-thread test added.